### PR TITLE
Artifactory-cli-go Version 1.2.0

### PR DESCRIPTION
**Pull request by jainishshah17**
Command line interface for Artifactory.

### CODE REVIEW

File: Library/Formula/artifactory-cli-go.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class ArtifactoryCliGo < Formula
+  desc "Provides a command-line interface."
+  homepage "https://www.jfrog.com"
```
Comment (mikemcquaid) [inline Library/Formula/artifactory-cli-go.rb]: What's the homepage for this actual project?

File: Library/Formula/artifactory-cli-go.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GOBIN"] = buildpath
+    ENV["GO15VENDOREXPERIMENT"] = "1"
+    (buildpath/"src/github.com/JFrogDev/artifactory-cli-go/").install Dir["*"]
+    system "go", "build", "-v", "github.com/JFrogDev/artifactory-cli-go/art/"
+    bin.install "art"
+  end
+  test do
+    system "#{bin}/art", "-v"
```
Comment (mikemcquaid) [inline Library/Formula/artifactory-cli-go.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/artifactory-cli-go.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class ArtifactoryCliGo < Formula
+  desc "Provides a command-line interface."
+  homepage "https://www.jfrog.com"
+  url "https://github.com/JFrogDev/artifactory-cli-go/archive/1.2.0.tar.gz"
+  sha256 "173a24e0b821c6ff2bb85fa4cbe733fa681a8284029b77387baf0fa5f2fcfc4e"
+  depends_on "go"
+  def install
```
Comment (mikemcquaid) [inline Library/Formula/artifactory-cli-go.rb]: Add a newline above this line.

File: Library/Formula/artifactory-cli-go.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class ArtifactoryCliGo < Formula
+  desc "Provides a command-line interface."
+  homepage "https://www.jfrog.com"
+  url "https://github.com/JFrogDev/artifactory-cli-go/archive/1.2.0.tar.gz"
+  sha256 "173a24e0b821c6ff2bb85fa4cbe733fa681a8284029b77387baf0fa5f2fcfc4e"
+  depends_on "go"
```
Comment (mikemcquaid) [inline Library/Formula/artifactory-cli-go.rb]: Add a newline above this line.

File: Library/Formula/artifactory-cli-go.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GOBIN"] = buildpath
+    ENV["GO15VENDOREXPERIMENT"] = "1"
+    (buildpath/"src/github.com/JFrogDev/artifactory-cli-go/").install Dir["*"]
+    system "go", "build", "-v", "github.com/JFrogDev/artifactory-cli-go/art/"
+    bin.install "art"
+  end
+  test do
```
Comment (mikemcquaid) [inline Library/Formula/artifactory-cli-go.rb]: Add a newline above this line.

File: Library/Formula/artifactory-cli-go.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+class ArtifactoryCliGo < Formula
+  desc "Artifactory-CLI provides a command-line interface."
+  homepage "https://github.com/JFrogDev/artifactory-cli-go"
+  url "https://github.com/JFrogDev/artifactory-cli-go/archive/1.2.0.tar.gz"
+  sha256 "173a24e0b821c6ff2bb85fa4cbe733fa681a8284029b77387baf0fa5f2fcfc4e"
+
+  depends_on "go"
```
Comment (DomT4) [inline Library/Formula/artifactory-cli-go.rb]: Is this only buildtime or needed at run as well?

Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: just build time.

Comment (DomT4) [inline Library/Formula/artifactory-cli-go.rb]: Should be `=> :build` then :)

File: Library/Formula/artifactory-cli-go.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GOBIN"] = buildpath
+    ENV["GO15VENDOREXPERIMENT"] = "1"
+    (buildpath/"src/github.com/JFrogDev/artifactory-cli-go/").install Dir["*"]
+    system "go", "build", "-v", "github.com/JFrogDev/artifactory-cli-go/art/"
+    bin.install "art"
```
Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: You can avoid this line by adding `"-o", "#{bin}/art"` to the `go build` line.

File: Library/Formula/artifactory-cli-go.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  test do
+    actual = pipe_output("#{bin}/art upload")
+    expected = "The --url option is mandatory\n"
+    assert_equal expected, actual
```
Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: Can’t we write a test that’s not an error?

File: Library/Formula/artifactory-cli-go.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
+class ArtifactoryCliGo < Formula
+  desc "Artifactory-CLI provides a command-line interface."
```
Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: Please remove `Artifactory-CLI` from the beginning of the description: `desc "command-line interface to ...`

File: Library/Formula/artifactory-cli-go.rb (hunk 7 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: If its not necessary I am ok with that.

File: Library/Formula/artifactory-cli-go.rb (hunk 8 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: This CLI has only three functionality, config, upload and download.

File: Library/Formula/artifactory-cli-go.rb [hunk 10]
```diff
@@ -0,0 +1,23 @@
+class ArtifactoryCliGo < Formula
+  desc "Provides a command-line interface."
```
Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: Which application/service is it a command-line for?

File: Library/Formula/artifactory-cli-go.rb (hunk 7 above)
Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: We use `bin.install "..."` when upstream doesn’t provide any way to install it automatically. Using `"-o", "#{bin}/art"` we can avoid this step and avoid a couple unnecessary disk operations.

File: Library/Formula/artifactory-cli-go.rb (hunk 8 above)
Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: Could we e.g. download a small resource?

File: Library/Formula/artifactory-cli-go.rb (hunk 10 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: Artifactory

File: Library/Formula/artifactory-cli-go.rb (hunk 8 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: it needs to have artifactory server up and running.

File: Library/Formula/artifactory-cli-go.rb (hunk 7 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: can you help me in making this change. I am not getting what exactly I have to change here.

Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: Remove `bin.install "art"` and change this line:
```rb
system "go", "build", "-v", "github.com/JFrogDev/artifactory-cli-go/art/"
```

into this one:
```rb
system "go", "build", "-o", "#{bin}/art", "-v", "github.com/JFrogDev/artifactory-cli-go/art/"
```

File: Library/Formula/artifactory-cli-go.rb [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+  desc "Provides a command-line interface for Artifactory."
+  homepage "https://github.com/JFrogDev/artifactory-cli-go"
+  url "https://github.com/JFrogDev/artifactory-cli-go/archive/1.2.0.tar.gz"
+  sha256 "173a24e0b821c6ff2bb85fa4cbe733fa681a8284029b77387baf0fa5f2fcfc4e"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GOBIN"] = buildpath
```
Comment (bfontaine) [inline Library/Formula/artifactory-cli-go.rb]: Why is it necessary to set `GOBIN` here?

File: Library/Formula/artifactory-cli-go.rb (hunk 7 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: got it thnx.

File: Library/Formula/artifactory-cli-go.rb (hunk 11 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: I will remove GOBIN. its unnacessary

File: Library/Formula/artifactory-cli-go.rb (hunk 8 above)
Comment (jainishshah17) [inline Library/Formula/artifactory-cli-go.rb]: is there anything else I need to change?

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/b0913d9ebc7c41218b8553de78bbdb3f4e720f98. Thank you for your contribution to Homebrew @jainishshah17; we appreciate it! :santa:

**Pull request closed without merging**